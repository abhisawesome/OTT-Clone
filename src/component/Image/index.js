import { memo,useCallback,useState } from 'react';
import PropTypes from 'prop-types';

const ImageBlock = ({
    alt = '',
    className = '',
    src = '',
    otherProps = {},
}) => {
    const [imageSrc,setImageSrc] = useState(src);
    
    // set the image on error
    const onImageError = useCallback( () => {
         setImageSrc('images/placeholder_for_missing_posters.png');
    },[src,imageSrc]);
    
    return (
        <img
            onError={onImageError}
            alt={alt}
            className={className}
            src={imageSrc}
            {...otherProps}
        />
    )
}

ImageBlock.propTypes = {
    alt: PropTypes.string,
    className: PropTypes.string,
    src: PropTypes.string,
    otherProps : PropTypes.object,
}
export default memo(ImageBlock);