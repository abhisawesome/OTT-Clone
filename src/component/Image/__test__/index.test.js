import renderer from 'react-test-renderer';
import Image from '../index';


describe('Image block',()=>{
    test('Snapshot testing',()=>{
        const tree = renderer.create(<Image 
            alt='alternative'
            className='new-class'
            src='src/image.jpg'
            key={1}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})
