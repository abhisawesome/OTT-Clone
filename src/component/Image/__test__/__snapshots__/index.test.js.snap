// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image block Snapshot testing 1`] = `
<img
  alt="alternative"
  className="new-class"
  onError={[Function]}
  src="src/image.jpg"
/>
`;
